### PR TITLE
[feature] more informative test output

### PR DESCRIPTION
**Pull request by adsteel**
Addresses issue https://github.com/brigade/scss-lint/issues/655

scss_output

### CODE REVIEW

**Comment by adsteel** [comment]
Outdated commit with original conversation [is here](https://github.com/adsteel/scss-lint/commit/da3ee26fe2e9f5926ef230ddf3cb55a7aaafccbd)

**Comment by connorshea** [comment]
Would really like this merged, one major thing I feel is missing from the gem right now :)

**Comment by sds** [comment]
Having a closer look at this I'm afraid this can't be merged as-is.

This results in this output being displayed for all different formatters, not just the default formatter. Thus it will break other formatters such as the JSON formatter since you're appending non-JSON to the end of the output.

You'll need to do some refactoring to have this happen in the `DefaultFormatter` class. I'm definitely in support of making SCSS-Lint's default output better, but we can't be breaking other output formats in the process. It needs to be isolated to the `DefaultFormatter`.

**Comment by lencioni** [comment]
Yeah, that's what I meant by https://github.com/adsteel/scss-lint/commit/da3ee26fe2e9f5926ef230ddf3cb55a7aaafccbd#commitcomment-15776473

**Comment by adsteel** [comment]
@sds Just getting back to this now, but I can't find a `DefaultFormatter` class.

**Comment by adsteel** [comment]
@sds @lencioni I've updated the code to log in the default reporter, though I'm still logging each individual linting as a period in the runner. I'm not familiar at all with the ecosystem, however. Does this work?

File: lib/scss_lint/runner.rb [hunk 1]
```diff
@@ -15,9 +17,10 @@ def initialize(config)
     # @param files [Array<Hash>] list of file object/path hashes
     def run(files)
       @files = files
-      @files.each do |file|
+      result = @files.each do |file|
```
Comment (sds) [inline lib/scss_lint/runner.rb]: What are you trying to do here? `result` will be set to `@files` since you're using `each`...

**Comment by sds** [comment]
Hey @adsteel,

No, this does not work. If I run `scss-lint --format JSON`, I get output like the following:

```
..................................................{
  "test.scss": [
    {
      "line": 1,
      "column": 2,
      "length": 140,
      "severity": "warning",
      "reason": "Variable names should be followed immediately by a colon",
      "linter": "SpaceAfterVariableName" 
    }
  ]
}
```

...which will obviously break any tools that rely on this format ([Overcommit](https://github.com/brigade/overcommit) being one of them).

This needs to support writing output to a separate stream (STDERR) so that the output format isn't broken.

**Comment by adsteel** [comment]
@sds Okay, I'll take one more swing at this on Friday. Without more knowledge of the ecosystem, or tests that define the API spec, this is a little like groping around in the dark.

**Comment by adsteel** [comment]
I wasn't able to come up with a solution I liked. I may re-attempt this at some point if someone else doesn't solve it, but in the meantime I need to stop working on it.

**Pull request closed without merging**